### PR TITLE
Created the POST endpoint for courses

### PR DESCRIPTION
**Pull request by abdulqshabbir**
API now accepts POST requests on `/courses` and will save a course to the database.

### CODE REVIEW

File: server/src/controllers/courses/createCourse.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import { Request, Response } from "express";
+
+export default async function createCourse(req: Request, res: Response) {
```
Comment (amirfakhrullah) [inline server/src/controllers/courses/createCourse.ts:5]: - When creating a course, we should store the `_id` of the course in the `user.courses` arrays
- It should look something like this:
```typescript
export default async function createCourse(req: Request, res: Response) {
   const id = getUserId(req); // helper to extract user id from the session
   const user = await User.findOne({ _id: id });
   
   // some error handling if no user found in db

    try {
        const newCourse = await createCourseInDB(req.body)
        user.courses.push(newCourse._id);
        await user.save();
        return res.status(200).json({ // also make sure the response to be consistent
             success: true,
             data: newCourse
         });
    } catch(e) {
        return res.status(400).json({
            success: false,
            message: e.message
        })
    }
}
```

you can refactor and separate them into services if you want, as long as the behavior is the same as above then it should be fine

**Comment by amirfakhrullah** [comment]
overall looks good :)

File: server/src/controllers/courses/createCourse.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+async function createCourseInDB(body: CoursePayloadType) {
+    try {
+        const newCourse = new Course({
+            name: body.name,
+            start_date: body.start_date,
+            end_date: body.end_date,
+            color: body.color,
+            weekly_schedule: body.weekly_schedule
+        })
```
Comment (AllMikeNoIke) [inline server/src/controllers/courses/createCourse.ts]: i'd use object destructuring in the parameters here, just to make the code a little cleaner

File: server/src/controllers/courses/createCourse.ts (hunk 1 above)
Comment (AllMikeNoIke) [reply server/src/controllers/courses/createCourse.ts:5]: it should be a 201 status code for successfully created content, as well!

Comment (abdulqshabbir) [reply server/src/controllers/courses/createCourse.ts:5]: Thanks! good call :)

File: server/src/controllers/courses/createCourse.ts (hunk 2 above)
Comment (abdulqshabbir) [reply server/src/controllers/courses/createCourse.ts]: Sounds good will make the changes!

**Review by amirfakhrullah — APPROVED** [review]

**Pull request merged**